### PR TITLE
Add use_pre_release for pypi source

### PR DESCRIPTION
**Pull request by BruceZhang1993**
Features
- Use pre release version for pypi source

Commits
- pypi: use_pre_release support

Signed-off-by: Bruce Zhang <zttt183525594@gmail.com>

### CODE REVIEW

File: nvchecker/source/pypi.py [hunk 1]
```diff
@@ -1,15 +1,25 @@
...
-get_version, get_cacheable_conf = simple_json(
-  PYPI_URL,
-  'pypi',
-  _version_from_json,
-)
+  headers = {
+    'Accept': 'application/json',
+    'User-Agent': 'lilydjwg/nvchecker'
+  }
+  url = 'https://pypi.org/pypi/{}/json'.format((package))
```
Comment (lilydjwg) [inline nvchecker/source/pypi.py]: double parentheses.

File: nvchecker/source/pypi.py [hunk 2]
```diff
@@ -1,15 +1,25 @@
...
+  use_pre_release = conf.getboolean('use_pre_release', False)
 
-get_version, get_cacheable_conf = simple_json(
-  PYPI_URL,
-  'pypi',
-  _version_from_json,
-)
+  headers = {
+    'Accept': 'application/json',
+    'User-Agent': 'lilydjwg/nvchecker'
```
Comment (lilydjwg) [inline nvchecker/source/pypi.py]: missing trailing comma.

File: nvchecker/source/pypi.py [hunk 3]
```diff
@@ -1,15 +1,25 @@
...
+  }
+  url = 'https://pypi.org/pypi/{}/json'.format((package))
+
+  async with session.get(url) as res:
+    data = await res.json()
+
+  if use_pre_release:
+    version = list(data['releases'].keys())[-1]
```
Comment (lilydjwg) [inline nvchecker/source/pypi.py]: Are dict keys in the same order of the JSON document? I don't think so, then which is the last one?

Comment (lilydjwg) [reply nvchecker/source/pypi.py]: You can explicitely sort the keys.

Comment (BruceZhang1993) [reply nvchecker/source/pypi.py]: Last commit passed CI successfully. And `sorted` is okay to sort version strings.

**Comment by lilydjwg** [comment]
2e55a7db60b5134dc83198f34fe3c42196bf7c1e

**Pull request closed without merging**